spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -396,7 +397,15 @@ public void close() throws IOException {
       // schema or rows returned by readers
       Schema finalSchema = expectedSchema;
       PartitionSpec spec = task.spec();
-      Set<Integer> idColumns = spec.identitySourceIds();
+
+      Set<Integer> idColumns = Sets.newHashSet();
```

Comment:
can you check if this adds latency for very large schemas? Since this is order of magnitude of the size of schema.